### PR TITLE
Update maintenance mode message

### LINKED ISSUE #1668: Update maintenance message
**Is your feature request related to a problem? Please describe.**
The current maintenance mode message is not very clear. It is not possible to notify users of upcoming maintenance.

**Describe the solution you'd like**
- An ENV variable to be set at deployment
- A message displayed regarding upcoming maintenance window
- Link to Discord while maintenance mode is enabled allowing users to easily follow updates.

### PR DESCRIPTION
**Pull request by FSM1**
Closes #1668 

This PR introduces an additional ENV variable `REACT_APP_MAINTENANCE_TIMESTAMP` which can be set to the proposed time of maintenance (Unix timestamp).

image
image

### CODE REVIEW

File: packages/files-ui/src/Components/Modules/LoginModule/InitialScreen.tsx [hunk 1]
```diff
@@ -171,6 +180,10 @@ const InitialScreen = ({ className }: IInitialScreen) => {
   const [error, setError] = useState<string | undefined>()
   const [errorEmail, setErrorEmail] = useState("")
   const maintenanceMode = process.env.REACT_APP_MAINTENANCE_MODE === "true"
+  const maintenanceWindowTimestamp = (process.env.REACT_APP_MAINTENANCE_TIMESTAMP &&
+    process.env.REACT_APP_MAINTENANCE_TIMESTAMP !== "")
+    ? Number(process.env.REACT_APP_MAINTENANCE_TIMESTAMP)
+    : undefined
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/LoginModule/InitialScreen.tsx]: We can cast it to number, and check if it's falsy then later.
```suggestion
  const maintenanceWindowTimestamp = Number(process.env.REACT_APP_MAINTENANCE_TIMESTAMP)
```

File: packages/files-ui/src/Components/Modules/LoginModule/InitialScreen.tsx [hunk 2]
```diff
@@ -461,6 +469,25 @@ const InitialScreen = ({ className }: IInitialScreen) => {
...
+                <Typography className={clsx(classes.maintenanceMessage, classes.maintenanceActiveMessage)}>
+                  <Trans>We are performing routine maintenance of the system. Service status updates will be posted on the{" "}
+                    <a href={ROUTE_LINKS.DiscordInvite}
+                      target="_blank"
+                      rel='noreferrer noopener'>Files Support Channel</a>{" "}
+                      on Discord
+                  </Trans>
+                </Typography>
+              )}
+              {!maintenanceMode && maintenanceWindowTimestamp && dayjs.unix(maintenanceWindowTimestamp).isAfter(dayjs()) && (
```
Comment (Tbaut) [inline packages/files-ui/src/Components/Modules/LoginModule/InitialScreen.tsx]:
```suggestion
              {!maintenanceMode && !!maintenanceWindowTimestamp && dayjs.unix(maintenanceWindowTimestamp).isAfter(dayjs()) && (
```

File: packages/storage-ui/src/Components/Modules/LoginModule.tsx [hunk 3]
```diff
@@ -170,7 +179,10 @@ const LoginModule = ({ className }: IInitialScreen) => {
   const { storageApiClient } = useStorageApi()
   const [email, setEmail] = useState("")
   const [errorEmail, setErrorEmail] = useState("")
-
+  const maintenanceWindowTimestamp = (process.env.REACT_APP_MAINTENANCE_TIMESTAMP &&
+    process.env.REACT_APP_MAINTENANCE_TIMESTAMP !== "")
+    ? Number(process.env.REACT_APP_MAINTENANCE_TIMESTAMP)
+    : undefined
```
Comment (Tbaut) [inline packages/storage-ui/src/Components/Modules/LoginModule.tsx]:
```suggestion
  const maintenanceWindowTimestamp = Number(process.env.REACT_APP_MAINTENANCE_TIMESTAMP)
```

File: packages/storage-ui/src/Components/Modules/LoginModule.tsx [hunk 4]
```diff
@@ -457,6 +464,25 @@ const LoginModule = ({ className }: IInitialScreen) => {
...
+                <Typography className={clsx(classes.maintenanceMessage, classes.maintenanceActiveMessage)}>
+                  <Trans>We are performing routine maintenance of the system. Service status updates will be posted on the{" "}
+                    <a href={ROUTE_LINKS.DiscordInvite}
+                      target="_blank"
+                      rel='noreferrer noopener'>Files Support Channel</a>{" "}
+                      on Discord
+                  </Trans>
+                </Typography>
+              )}
+              {!maintenanceMode && maintenanceWindowTimestamp && dayjs.unix(maintenanceWindowTimestamp).isAfter(dayjs()) && (
```
Comment (Tbaut) [inline packages/storage-ui/src/Components/Modules/LoginModule.tsx]:
```suggestion
              {!maintenanceMode && !!maintenanceWindowTimestamp && dayjs.unix(maintenanceWindowTimestamp).isAfter(dayjs()) && (
```

**Review by Tbaut — COMMENTED** [review]
Working well on Files. 

Submitted some code suggestions to make things a little easier. The one thing I find not super user friendly is that we talk about a maintenance window, but the users have no idea about how long this window is. I'd just say "the system will be unavailable at this time" or talk about a short moment. The window sounds like we know how long it'll be, but we won't tell it :D

**Review by asnaith — APPROVED** [review]
Works as intended when the flag is set in the .env file. 

My only suggestion would be to perhaps add more keys to the `.env.example`. Not knowing about REACT_APP_MAINTENANCE_MODE=true confused me for a bit whilst testing this :sweat_smile:

**Review by Tbaut — APPROVED** [review]
:tada:

**Pull request merged**